pymatgen/analysis/spectra/spectrum.py

```diff
@@ -0,0 +1,135 @@
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+from monty.json import MSONable
+import six
+from pymatgen.core.units import EnergyArray, Energy
+from scipy.interpolate import interp1d
+from pymatgen import Structure
+import numpy as np
+
+"""
+This module defines classes to represent all spectra
+"""
+
+__author__ = "Chen Zheng"
+__copyright__ = "Copyright 2012, The Materials Project"
+__version__ = "2.0"
+__maintainer__ = "Chen Zheng"
+__email__ = "chz022@ucsd.edu"
+__date__ = "Aug 9, 2017"
+
+
+class Spectrum(MSONable):
+    """
+    Base class for Xane spectrum, Exaf spectrum, NMR.
+    Not meant to be instantiated directly
+    """
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def intensity_sum_norm(self):
+        """
+        Normalize the spectrum with respect to the sum of intensity
+        :return:
+        """
+        self.y = self.y / np.sum(self.y)
+
+    def get_smeared_density(self, sigma):
+        """
+        Return Gaussian smeared spectrum
+        :param sigma: Std dev of Gaussian smearing function
+        :return: Gaussian-smeared densities
+        """
+
+        from scipy.ndimage.filters import gaussian_filter1d
+        diff = [self.x[i + 1] - self.x[i] for i in range(len(self.x) - 1)]
+        avg_eV_per_step = np.sum(diff) / len(diff)
+        rv = gaussian_filter1d(self.y, sigma / avg_eV_per_step).tolist()
+        return rv
+
+    def ensemble_average(self, spect_list, spect_multi):
```

Comment:
This is not what you are supposed to do. Look at how DOS does it. You just need to implement `__add__` and `__mult__` and you automatically can do linear combination of spectrum. E.g., A *spec1 + B * spec2. That will automatically allow you to do ensemble averages. 

I said many times - LOOK at the DOS object. I don't know how many times I have to write this.